doc/news/_preparation_next_release.md

```diff
@@ -281,11 +285,25 @@ This section keeps you up-to-date with the multi-language support provided by El
 
 ## Tools
 
-### <<Tool>>
+### KDB
 
+- add basic C framework, helper functions and new `get` implementation _(@hannes99)_
+  - Logging levels and error/warning outputs are now handled my the new CLI framework, and not by commands individually.
+  - Using `opts` for command spec allows automatic generation of help messages and handles all the parsing.
+  - The `--help` flag will now show a description of how to use the command, instead of opening the man page.
+  - Not providing the correct options or arguments the commands expect will now result in a short error message describing what is wrong,
+    this is generated my `opts`.
 - <<TODO>>
+- add spec for C++ commands _(@hannes99)_
+  - Add specification for commands written in `C++`, this enables `opts` to generate the usage message and makes sure all commands(
+    regardless of their implementation) have the same help message format and generally behave the same.
+  - _NOTE_: The `path` argument for `kdb complete` is now required, so instead of `kdb complete` `kdb complete ""` has to be used. The
+    reason for this is described in issue #4952.
 - <<TODO>>
-- <<TODO>>
+- add support for external commands, with and without spec _(@hannes99)_
+  - It is still possible to execute external binaries that are placed in a specific director, same as it worked in the old implementation.
+  - It is now possible to provide a specification to external commands to `kdb`, so `kdb` check/parses arguments and then calls the
+    external program with them.
```

Comment:
Please show how this can be done.

```suggestion
  - It is now possible to provide a specification to external commands to `kdb`, so `kdb` check/parses arguments and then calls the external program with them.
```

Reply:
Added link to the tutorial